planner/filter/normalize.go

```diff
@@ -185,7 +185,12 @@ func normalizeProperties(parentKey connor.FilterKey, conditions []any) []any {
 	// if canMergeAnd is true, all _and groups will be merged
 	props := make(map[int][]any)
 	for _, c := range conditions {
-		for key, val := range c.(map[connor.FilterKey]any) {
+		cMap, ok := c.(map[connor.FilterKey]any)
```

Comment:
question: Why has this changed?  What else could it be?

Reply:
`normalizeProperty` method that calls this `normalizeProperties` checks is given condition is `map[connor.FilterKey]any` or an array of conditions which has a type of `[]any`. But as it recursively checks all nested conditions and values one of the values (not conditions) can match `[]any` (for example "_in" op). That's why we need to check if it's really a conditions or just value.

Reply:
Ah okay - this was a bug fix?  Did you add an integration test that covers this?

Reply:
an integration test found this bug :)